src/task/ReceiveVideoStreamIndexTask.ts

```diff
@@ -74,13 +75,38 @@ export default class ReceiveVideoStreamIndexTask extends BaseTask
       videoUplinkBandwidthPolicy,
     } = this.context;
 
+    const oldAttendees = videoStreamIndex.allVideoSendingAttendeesExcludingSelf(selfAttendeeId);
     videoStreamIndex.integrateIndexFrame(indexFrame);
     videoDownlinkBandwidthPolicy.updateIndex(videoStreamIndex);
     videoUplinkBandwidthPolicy.updateIndex(videoStreamIndex);
 
     this.resubscribe(videoDownlinkBandwidthPolicy, videoUplinkBandwidthPolicy);
     this.updateVideoAvailability(indexFrame);
     this.handleIndexVideosPausedAtSource();
+    const newAttendees = videoStreamIndex.allVideoSendingAttendeesExcludingSelf(selfAttendeeId);
+    if (!this.areVideoSendingAttendeesEqual(oldAttendees, newAttendees)) {
+      this.context.audioVideoController.forEachObserver((observer: AudioVideoObserver) => {
+        Maybe.of(observer.remoteVideosAvailableDidChange).map(f => f.bind(observer)(newAttendees));
+      });
+    }
+  }
+
+  private areVideoSendingAttendeesEqual(
+    oldAttendees: VideoSendingAttendee[],
+    newAttendees: VideoSendingAttendee[]
+  ): boolean {
+    const compare = (attendeeA: VideoSendingAttendee, attendeeB: VideoSendingAttendee): number =>
+      attendeeA.attendeeId.localeCompare(attendeeB.attendeeId);
+
+    oldAttendees.sort(compare);
```

Comment:
'Array.sort' sorts an array in place. The `newAttendees` value in `handleIndexFrame()` as well as in the builder's observer method.

A less error-prone option is to clone it first.

```
const sortedOldAttendees = [...oldAttendees].sort(compare);
const sortedNewAttendees = [...newAttendees].sort(compare);
```

Reply:
Ok ok got it, fixed this. Thanks!